supportedBackends/aws/src/main/scala/cromwell/backend/impl/aws/AwsAsyncJobExecutionActor.scala

```diff
@@ -0,0 +1,255 @@
+              // Log exceptions and return the original handle to try again.
+              jobLogger.warn(s"Caught exception, retrying", exception)
+              oldHandle
+          }
+        handler(exception)
+      case error: Error => throw error // JVM-ending calamity.
+      case _: Throwable =>
+        // Someone has subclassed or instantiated Throwable directly. Kill the job. They should be using an Exception.
+        FailedNonRetryableExecutionHandle(throwable)
+    }
+  }
+
+  private def tellMetadata(metadataKeyValues: Map[String, Any]): Unit = {
+    import cromwell.services.metadata.MetadataService.implicits.MetadataAutoPutter
+    serviceRegistryActor.putMetadata(jobDescriptor.workflowDescriptor.id, Option(jobDescriptor.key), metadataKeyValues)
+  }
+
+  private def executionResult(status: BackendRunStatus, handle: BackendPendingExecutionHandle)
+                             (implicit ec: ExecutionContext): ExecutionHandle = {
+    try {
+      if (isSuccess(status)) {
+
+        lazy val stderrLength: Long = File(remoteStrErrPath).size
+        lazy val returnCode: Try[Int] = returnCodeContents.map(_.trim.toInt)
+        status match {
+          case _ if failOnStderr && stderrLength.intValue > 0 =>
+            // returnCode will be None if it couldn't be downloaded/parsed, which will yield a null in the DB
+            FailedNonRetryableExecutionHandle(new RuntimeException(
+              s"execution failed: stderr has length $stderrLength"), returnCode.toOption)
+          case _ if returnCodeContents.isFailure =>
+            val exception = returnCode.failed.get
+            jobLogger.warn(s"could not download return code file, retrying", exception)
+            // Return handle to try again.
+            handle
+          case _ if returnCode.isFailure =>
+            FailedNonRetryableExecutionHandle(new RuntimeException(
+              s"execution failed: could not parse return code as integer: ${returnCodeContents.get}"))
+          case _ if !continueOnReturnCode.continueFor(returnCode.get) =>
+            val badReturnCodeMessage = s"Call ${jobDescriptor.key}: return code was ${returnCode.getOrElse("(none)")}"
+            FailedNonRetryableExecutionHandle(new RuntimeException(badReturnCodeMessage), returnCode.toOption)
+          case _ =>
+            handleSuccess(status, handle, returnCode.get)
+        }
+
+      } else {
+        handleFailure(status, handle)
+      }
+    } catch {
+      case e: Exception =>
+        jobLogger.warn("Caught exception processing job result, retrying", e)
+        // Return the original handle to try again.
+        handle
+    }
+  }
+
+  private lazy val instantiatedCommand = Command.instantiate(
+    jobDescriptor, commandLineFunctions, commandLinePreProcessor, commandLineValueMapper).get
+
+  private lazy val tag = s"${this.getClass.getSimpleName} [UUID(${workflowId.shortString}):${jobDescriptor.key.tag}]"
+
+  // aws backend specific
+
+  type BackendJobId = AwsJobId
```

Comment:
are these types because they're that way in a parent trait/class? If not, use case classes so the compiler can actually do something about the type information